algorithmia/src/kafka-datadog.py

```diff
@@ -0,0 +1,61 @@
+import os
+import json
+import sys
+from datadog import initialize, api
+from kafka import KafkaConsumer
+
+try:
```

Comment:
Just a question, is there any reason we don't want to this to be a check that runs with an agent?

Reply:
Adding Michael's explanation here based on our call with Kris today. 

Algorithmia would like to read from the kafka streams directly which is a firehose of data and isn't well suited for a periodic check, hence this direct to API approach. We highlighted the alternative to use DogStatsD with a local Agent install to get some of the agent benefits, but that will likely be a future evolution for Algorithmia. 

Thanks!

Reply:
Thanks, @alexvetras, that is correct. I prototyped an approach using an agent check before submitting this version, and because hooking into a Kafka pipeline is a blocking call rather than a finite check, it hangs when the agent check runs, hence the approach used here (and the plans for future work that Alex mentioned for sending this data to DogStatsD).

Reply:
If this is not an agent check then the python part should not be in this repo (owned by algorithmia?). That would also have the advantage for the users of not having to clone this entire repo to run it and for you to not depend on code reviews from Datadog's side

Reply:
As we continue to work with customers on this integration, it is possible that this evolves into a script with custom metrics getting sent into DogStatsD. Also, having the Python message forwarding component here ensures a tighter coupling between the payload from Algorithmia and the default metrics that are configured in the dashboard and monitor in terms of the prefix, tags, etc. So, I am okay leaving the message forwarding component in this repo in the meantime if that is acceptable.

Reply:
>  it is possible that this evolves into a script with custom metrics getting sent into DogStatsD

I understand that a DogStatsD emitter would also be an external process running separately from the agent so it would still not be an agent integration.
Hosting the code on this repo comes with following advantages, none of those apply to a separate process:
* CI set up (this has no tests)
* Validation to ensure it follows guidelines for developing agent checks (this is not a custom check)
* Wheel releases so integrations can be installed directly from the agent using a command (it would not apply since its not an agent check)
With the disadvantage of not having merge ownership

> Also, having the Python message forwarding component here ensures a tighter coupling between the payload from Algorithmia and the default metrics that are configured in the dashboard and monitor in terms of the prefix, tags, etc

That coupling is virtual because there's no automation that ensures it (versus on agent checks where automated testing can be added to ensure metadata.csv matches the actual metrics etc).

All the other datadog or dogstastd emitters that have integration tiles host the code themselves (and it's often part of the product to monitor itself rather than a separate script).

Reply:
Got it, thanks for clarifying! Should I go ahead and move the Python component out of this repo and update the documentation here?

Reply:
Yes, please

Reply:
Thanks! In 7604a17, I moved the Python script and specific steps to our integrations repo at https://github.com/algorithmiaio/integrations/tree/master/Datadog, and I updated the documentation here to reflect the update.